w3af/plugins/crawl/url_fuzzer.py

```diff
@@ -269,6 +314,33 @@ def _mutate_file_type(self, url):
                 url_copy.set_extension(filetype)
                 yield url_copy
 
+    def _mutate_file_name(self, url):
+        filename = url.get_file_name()
+        if filename:
+            domain = url.get_domain_path()
+            url_string = domain.url_string
+            name = filename[:filename.rfind(u'.')]
```

Comment:
What happens here when the input URL is `http://w3af.org/foobar`  (there is no point in the filename)

Reply:
See below's answer.